dotCMS/src/main/java/com/dotcms/graphql/business/ContentAPIGraphQLTypesProvider.java

```diff
@@ -0,0 +1,246 @@
+                }
+            }
+
+        });
+
+        return fieldDefinitions;
+    }
+
+    public GraphQLOutputType getGraphqlTypeForFieldClass(final Class<? extends Field> fieldClass,
+            final Field field) {
+        return fieldClassGraphqlTypeMap.get(fieldClass) != null
+                ? fieldClassGraphqlTypeMap.get(fieldClass)
+                : fieldClass.equals(TextField.class) && field.dataType().equals(DataTypes.INTEGER)
+                        ? GraphQLInt
+                        : fieldClass.equals(TextField.class) && field.dataType()
+                                .equals(DataTypes.FLOAT) ? GraphQLFloat
+                                : GraphQLString;
+    }
+
+    public DataFetcher getGraphqlDataFetcherForFieldClass(final Class<Field> fieldClass) {
+        return fieldClassGraphqlDataFetcher.get(fieldClass) != null
+                ? fieldClassGraphqlDataFetcher.get(fieldClass)
+                : new FieldDataFetcher();
+    }
+
+    /**
+     * This method determines whether a {@link Field}'s variable is compatible with the
+     * current GraphQL Schema.
+     *<p>
+     * The {@link Field} is deemed compatibly if any of the followings conditions are true:
+     * <ul>
+     *     <li>The field variable does not match any of the inherited fields names from the {@link InterfaceType#getContentletInheritedFields} </li>
+     *     <li>The field variable matches the name of a inherited field but neither of them have a {@link CustomFieldType#getCustomFieldTypes}
+     *     as its mapped GraphQL Type
+     * </ul>
+     * @param variable the variable whose compatibility will be checked
+     * @param field the field which the variable will be set to
+     * @return
+     */
+    public boolean isFieldVariableGraphQLCompatible(final String variable, final Field field) {
+        // first let's check if there's an inherited field with the same variable
+        if (InterfaceType.getContentletInheritedFields().containsKey(variable)) {
+            // now let's check if the graphql types are compatible
+
+            // get inherited field's graphql type
+            final GraphQLType inheritedFieldGraphQLType = InterfaceType
+                    .getContentletInheritedFields()
+                    .get(variable).getType();
+
+            // get new field's type
+            final GraphQLType fieldGraphQLType = getGraphqlTypeForFieldClass(field.type(), field);
+
+            // if at least one of them is a custom type, they need to be equal to be compatible
+            return (!isCustomFieldType(inheritedFieldGraphQLType)
+                    && !isCustomFieldType(fieldGraphQLType))
+                    || inheritedFieldGraphQLType.equals(fieldGraphQLType);
+        }
+
+        return true;
+    }
+
+    @VisibleForTesting
+    protected void setFieldGeneratorFactory(
+            GraphQLFieldGeneratorFactory fieldGeneratorFactory) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'fieldGeneratorFactory' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)